### PR TITLE
Test base

### PR DESCRIPTION
**Pull request by tmat**
Test only change (refactoring).

Currently many test classes across Roslyn compiles and services derive from TestBase. TestBase has become a kitchen sink of helpers. Some of them make sense for all tests, but a lot of them are specific to compilers and are not relevant to services. This change takes a step at separating compiler specific helpers, such as PDB validation to a separate class and project -- Roslyn.Test.PdbUtilities. Long term we could move more IL and metadata validation to this project and rename it to CompilerTestUtilities.

- Move CompilationVerifier out of TestBase class.
- Move all PDB validation helpers to PdbValidation class.
- Move PdbValidation class to Roslyn.Test.PdbUtilities project.

### CODE REVIEW

**Comment by tmat** [comment]
@dotnet/roslyn-compiler Please review -- this is test only change. Recommended to review each commit separately.

**Comment by tmat** [comment]
@ivanbasov @jinujoseph

**Comment by ivanbasov** [comment]
:shipit:

**Comment by tmat** [comment]
@dotnet/roslyn-compiler Anybody else interested in reviewing?

**Comment by jcouv** [comment]
Starting review.

File: src/Test/Utilities/Portable/CommonTestBase.CompilationVerifier.cs [hunk 1]
```diff
@@ -312,10 +297,32 @@ private IModuleSymbol GetModuleSymbolForEmittedImage(ImmutableArray<byte> peImag
...
                 var references = _compilation.References.Concat(new[] { targetReference });
-                var assemblies = _test.ReferencesToModuleSymbols(references, importOptions);
+                var assemblies = GetReferencesToModuleSymbols(references, importOptions);
                 return assemblies.Last();
             }
 
+            private IEnumerable<IModuleSymbol> GetReferencesToModuleSymbols(IEnumerable<MetadataReference> references, MetadataImportOptions importOptions)
+            {
+                var dummy = _compilation
+                    .RemoveAllReferences()
```
Comment (jcouv) [inline src/Test/Utilities/Portable/CommonTestBase.CompilationVerifier.cs]: RemoveAllReferences is just the same as WithReferences with some empty array. What is the benefit of calling WithReferences twice?

File: src/Test/Utilities/Portable/Compilation/CompilationDifference.cs [hunk 2]
```diff
@@ -3,35 +3,30 @@
...
 using Microsoft.Metadata.Tools;
 using Roslyn.Test.MetadataUtilities;
-using Roslyn.Test.PdbUtilities;
 using Roslyn.Test.Utilities;
-using Xunit;
 
 namespace Microsoft.CodeAnalysis.Test.Utilities
 {
-    internal sealed class CompilationDifference
+    public sealed class CompilationDifference
```
Comment (jcouv) [inline src/Test/Utilities/Portable/Compilation/CompilationDifference.cs]: Why `public`?

Comment (tmat) [inline src/Test/Utilities/Portable/Compilation/CompilationDifference.cs]: There is no need for it to be internal. Public is easier as it avoids more internals visible to between test assemblies.

**Comment by tmat** [comment]
@jcouv Thanks for review.

**Pull request merged**